forge/forge/verify/verify.py

```diff
@@ -254,7 +256,84 @@ def verify_golden(
     device: "TTDevice",
     inputs: Tuple[Tensor],
     outputs: Tuple[torch.Tensor],
-    verify_cfg: VerifyConfig,
+    verify_cfg: DepricatedVerifyConfig,
 ):
 
     assert False  # Run ttnn golden
+
+
+def verify(
+    inputs: List[Union[torch.Tensor, tf.Tensor, tf.Variable]],
+    framework_model: Union[torch.nn.Module, tf.Module, tf.keras.Model],
+    compiled_model: Any,
+    verify_cfg: VerifyConfig = global_verify_config,
+):
+    """
+    Verify the compiled model against the framework model
+    """
+    if not verify_cfg.enabled:
+        logger.warning("Verification is disabled")
+        return
+
+    from forge.op.eval.common import compare_with_golden  # avoid circular import
+
+    # 0th step: input checks
+
+    # Check if inputs are of the correct type
+    if not inputs:
+        raise ValueError("Input tensors must be provided")
+    for input_tensor in inputs:
+        if not isinstance(input_tensor, verify_cfg.supported_tensor_types):
+            raise TypeError(
+                f"Input tensor must be of type {verify_cfg.supported_tensor_types}, but got {type(input_tensor)}"
+            )
+
+    if not isinstance(framework_model, verify_cfg.framework_model_types):
+        raise TypeError(
+            f"Framework model must be of type {verify_cfg.framework_model_types}, but got {type(framework_model)}"
+        )
+
+    if not isinstance(compiled_model, verify_cfg.compiled_model_types):
+        raise TypeError(
+            f"Compiled model must be of type {verify_cfg.compiled_model_types}, but got {type(compiled_model)}"
+        )
+
+    # 1st step: run forward pass for the networks
+    fw_out = framework_model(*inputs)
+    co_out = compiled_model(*inputs)
+
+    # 2nd step: apply preprocessing (push tensors to cpu, perform any reshape if necessary,
+    #  cast from tensorflow tensors to pytorch tensors if needed)
+    if not isinstance(fw_out, torch.Tensor):
+        fw_out = to_pt_tensors(fw_out)
+
+    if not isinstance(co_out, torch.Tensor):
+        co_out = to_pt_tensors(co_out)
+
+    co_out = [co.to("cpu") for co in co_out]
+    fw_out = [fw_out] if isinstance(fw_out, torch.Tensor) else fw_out
+
+    # 3rd step: verifications of outputs
+    # - dtype check
+    # - shape check
+    # - compare with golden
+    if verify_cfg.verify_size:
+        assert len(fw_out) == len(
```

Comment:
Can we raise exceptions instead of doing asserts? Exceptions are a bit more descriptive, and easier to manage compared to assertions. 

This function has a mix of both. Do you have specific reason for choosing ether/or in this function?

Reply:
I've used `exceptions` for input errors (e.g. someone passing the model that we don't support). On the other hand, for errors caused by our compiler `assertions` are used. Should I raise exceptions for everything instead?

Reply:
Behavior is similar, but I'm more in favor of using Exceptions. Main reasons:
1. You have more granularity by exception type (either existing or custom exceptions)
2. Better control over them (e.g. custom logic when hit)
3. We can catch them during the program execution and try to do recovery if possible
4. They can't be ignored with specific Python optimizations (-O)